_meeting-notes/2019-12-10.md

```diff
@@ -0,0 +1,150 @@
+~~https://raw.githubusercontent.com/ipfs/testground/master/docs/img/Q4OKRs.png~~
+~~### Top 3 things to get done this week~~
+
+We are putting aside the OKRs for the time being and instead **focus on the #1 Goal: Unblock go-ipfs 0.5.0**. This goal has a very clear list of things that need to ship:
+
+- [ ] Instrument go-libp2p so that the DHT operations are observable and traced in a Distributed Context (owner: @raulk)
+- [ ] Add support to Traffic Shaping & Network Configuration to TestGround (owner: @Stebalien)
+- [ ] Expand the TestGround Infrastructure so that it can run multiple thousands of nodes (owner: @nonsense)
+- [ ] Creation of a TestGround Dashboard that can aid the developers of go-ipfs & go-libp2p to grok the results outputted by every node in a TestGround run (owner: @daviddias)
+- [ ] Keep on iterating the DHT Test Plan (@owner: @daviddias)
+- [ ] Design blueprint for multi-version tests (aka test ensembles) (owner: @raulk)
+
+Full list can be at https://github.com/ipfs/testground/issues/196
+
+
+
+### Assignments
+
+- ==@raulk==
+    - (HIGH PRIO, working on this now) Outline candidate solutions and trade-offs for libp2p deep traceability.
+    - Proof of concept of 3 different solutions that Raúl is thinking of (OpenTracing with mock tracer, diagnostics context, go-log tracer).
+- ==@daviddias==
+  - Host Testground Hack week
+  - Study the current situation of go-libp2p metrics and report back
+    - See how open-census is working there
+    - Experiment with jaeger 
+  - Review @jimpick PR https://github.com/ipfs/testground/pull/232
+- ==@nonsense==
+  - Measure the memory footprint of a go-libp2p node that runs 1000 find peer queries on the mainnet
+  - Use that measure to calculate how many nodes can be run in each AWS instance
+  - Calculate the right number of machines needed for a cluster to run 10000 libp2p nodes
+  - Enhance the Testground experience by removing the overhead of running a plan (devs should not have to worry about infra). Running a plan should be done in one command
+      - @nonsense - this assumes we run a service on AWS and solve issues like:
+          - https://github.com/ipfs/testground/issues/226
+  - Get filebeat to only pump logs at the end of a run (so that it doesn't influence the performance of the test)
+      - @nonsense - this seems to be the lowest priority for the week, right?
+- ==@hacdias==
+  - Keep working on Chewing Strategies (https://github.com/ipfs/testground/issues/93)
+      - Create an issue with the APIs that are missing from Core API and Go-ipfs-api
+      - Should go-ipfs-api and core api have the same interface?
+  - DHT refactory thingy (see @stebalien's) update bellow
+- ==@jimpick==
+  - Finish "brain dump" for Nodes Connectivity planning https://github.com/ipfs/testground/pull/232
+  - Basic libp2p connectivity test + NAT (needs to be broken into sub-tasks)
+  - Review current open bugs and tackle a set of them
```

Comment:
@jimpick here are some suggestions for you to take on this week:

- https://github.com/ipfs/testground/issues/191
- https://github.com/ipfs/testground/issues/173
- https://github.com/ipfs/testground/issues/122
- https://github.com/ipfs/testground/issues/34
- https://github.com/ipfs/testground/issues/54
- https://github.com/ipfs/testground/issues/118
- https://github.com/ipfs/testground/issues/124
- https://github.com/ipfs/testground/issues/131

Reply:
@jimpick from this list, please highlight the ones you expect to deliver this week.

Reply:
So, I stole some: #191 #124 and #158 :)

Reply:
I'm gonna steal #34 too.

Reply:
@hacdias is on 🔥🔥🔥🔥

@jimpick come before it is too late and @hacdias shipped them all 😁

Reply:
Hey, I was sleeping! I guess I get what is left over. :-)

Thoughts on the 5 remaining ones:

#173 Recreate-broken containers

I'll try to reproduce the failing scenario and do a quick fix.

#122 RunEnv.initLogger is racy

I'll take a look.

#54 Test the TestGround

Moving target, big scope. I labelled it as an "Epic". Needs to be broken down into smaller tasks that can be completed in a short time frame.

#118 Use go.mod for the build step

I think this is missing a design blueprint.

#131 Make Runner.Run() and Builder.Build() take contexts

I can try to add these.

Reply:
Ok, updating your assignments list